provider/datagen/src/bin/datagen.rs

```diff
@@ -49,26 +47,28 @@ fn main() -> eyre::Result<()> {
                 .possible_value("json")
                 .possible_value("bincode")
                 .possible_value("postcard")
-                .help("File format syntax for data files."),
+                .help("--format=dir only: serde serialization format."),
         )
         .arg(
             Arg::with_name("PRETTY")
                 .short("p")
                 .long("pretty")
-                .help("Whether to pretty-print the output files. Only affects JSON and Rust modules."),
+                .help("--format=mod, --format=dir only: pretty-print the Rust or JSON output files."),
         )
         .arg(
             Arg::with_name("FINGERPRINT")
                 .long("fingerprint")
-                .help("Whether to add a fingerprints file to the output. Not compatible with --format=blob")
+                .help("--format=dir only: whether to add a fingerprints file to the output.")
         )
         .arg(
             Arg::with_name("CLDR_TAG")
                 .short("t")
                 .long("cldr-tag")
                 .value_name("TAG")
+                .default_value("latest")
```

Comment:
### Discussion on setting defaults for `--cldr-tag` and `--icuexport-tag`

Pros and cons for making this change:

1. Pro: This is the option most people should choose if they don't know any better.
2. Con: It makes the default behavior of `icu4x-datagen` be non-deterministic and network-dependent, which seems undesirable.
3. Con: It's good for educational purposes for people to make an explicit decision about where to get their data.

Reply:
> Con: It makes the default behavior of icu4x-datagen be non-deterministic and network-dependent, which seems undesirable.

I see icu4x-datagen as a simple tool to get started with ICU4X, so it should be allowed to use the network by default. Note that the API does not download by default, which is more appropriate for build scripts.

> Con: It's good for educational purposes for people to make an explicit decision about where to get their data.

I don't think we should encourage non-standard or outdated data sources. If clients want to use the binary in some kind of build process, they will look at `icu4-datagen --help`.

Reply:
If others agree that network access in the CLI version of `icu4x-datagen` is OK as a default, I'm fine with this.

Reply:
@zbraniecki @Manishearth

Reply:
Merging this because it blocks another PR, we can address this in a follow-up.